### DISCUSSION TITLE
db.$reset() fails and IDK why

### DISCUSSION THREAD

**Discussion by Vandivier** [blitz-js/blitz#3058] [Q&A] [answered]
`resetPassword.test.ts` is failing for me on windows OOTB after updating from 0.44.0 to 0.44.3 on the beforeEach block:

```
 FAIL   SERVER  app/auth/mutations/resetPassword.test.ts (5.904 s)
  ● resetPassword mutation › works correctly

    Failed: 1

      3 | import { hash256, SecurePassword } from "blitz"
      4 |
    > 5 | beforeEach(async () => {
        | ^
      6 |   await db.$reset()
      7 | })
      8 |

      at Env.beforeEach (node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:46:24)
      at Object.<anonymous> (app/auth/mutations/resetPassword.test.ts:5:1)
```

I can functionally verify that resetting the password works fine at runtime. Has anyone run into this? I don't see much documentation on `db.$reset` or `db` in tests in the docs. I'm just using the OOTB sqlite db.

---

edit: I started a new project from scratch @0.44.3 and didn't have this issue. Maybe something to do with the upgrade?

**Comment by Vandivier** [comment, marked as answer]
I found the issue by running:
`blitz prisma migrate dev`

there was an issue in a migration file

It would be nice if there was some other way to float up this issue because I am in a [planetscale workflow](https://github.com/blitz-js/blitz/issues/3053) so we don't really run this command

anyway I nuked and reset my db after upgrading and the issue went away

**Comment by zipzapzanigan** [comment]
I also ran into this problem, and submitted a duplicate bug: https://github.com/blitz-js/blitz/issues/3180 
This would be a very easy fix in the definition of $reset. It already calls a system call to prisma, you could just simply 
- add a conditional to check the database type before running the command
- send a mysql command to drop the tables
- run prisma db push
